src/Support/Type.php

```diff
@@ -89,6 +89,9 @@ public function getFields(): array
                 $field = app($field);
                 $field->name = $name;
                 $allFields[$name] = $field->toArray();
+            } elseif ($field instanceof Field) {
+                $field->setName($name);
```

Comment:
Let's do it without the setter, as it's done 3 lines above => just `$field->name = $name`.

The reason is homogeneity:
- `\Rebing\GraphQL\Support\Field` implements `__set` exactly for this reason.
- there should be one "true" way of doing things
- introducing this would mean changing _all_ of the instances we've (refactor)

If this is so important, it should be done in separate PR

Reply:
Yeah I initially did that but CI (phpstan) was complaining. So I did it this way.

Will change back, add tests and docs first thing tomorrow. M

Reply:
Done. PHPstan still complaining though. The class has a `@property` annotation already, so not really sure why.

```
Line   src/Support/Type.php
 ------ ----------------------------------------------------------------------
  93     Access to an undefined property Rebing\GraphQL\Support\Field::$name.
```